File: ruby_trunk/logic/faculty.rb
require 'logic/course'

module TTime
  module Logic
    class Faculty
      attr_accessor :name, :courses

      def initialize(name, courses = [])
        @name = name
        @courses = courses
      end

      def get_course(course_number)
        @courses.each do |course|
          return course if course.number.to_i == course_number.to_i
        end
        nil
      end
    end
  end
end

File: src/com/ttime/logic/Faculty.java
package com.ttime.logic;

import java.util.HashSet;
import java.util.Set;

public class Faculty implements Comparable<Faculty> {
    String name;
    Set<Course> courses;

    public Set<Course> getCourses() {
        return courses;
    }

    public Faculty(String name) {
        this.name = name;
        this.courses = new HashSet<Course>();
    }

    public String getName() {
        return this.name;
    }

    @Override
    public String toString() {
        return this.name;
    }

    @Override
    public int compareTo(Faculty o) {
        return name.compareTo(o.getName());
    }
}
